repo_utils/parsed_diff.py

```
@@ -0,0 +1,368 @@
+        minus_count = 0
+
+        def _flush_segment() -> None:
+            nonlocal segment_new_start, plus_count, minus_count
+
+            if plus_count and segment_new_start is not None:
+                new_range = (segment_new_start, segment_new_start + plus_count - 1)
+                if minus_count:
+                    changed_ranges.append(new_range)
+                else:
+                    added_ranges.append(new_range)
+            elif minus_count and new_line > 0:
```

Ruff (0.15.10)

[warning] Function definition does not bind loop variable `new_line`

(B023)